sdk/pinocchio/src/sysvars/instructions.rs

```diff
@@ -0,0 +1,236 @@
+use crate::{
+    account_info::{AccountInfo, Ref},
+    instruction::AccountMeta,
+    program_error::ProgramError,
+    pubkey::Pubkey,
+    sanitize_error::SanitizeError,
+};
+
+use core::{marker::PhantomData, mem::size_of, ops::Deref};
+
+/// Sysvar1nstructions1111111111111111111111111
+pub const INSTRUCTIONS_ID: Pubkey = [
+    0x06, 0xa7, 0xd5, 0x17, 0x18, 0x7b, 0xd1, 0x66, 0x35, 0xda, 0xd4, 0x04, 0x55, 0xfd, 0xc2, 0xc0,
+    0xc1, 0x24, 0xc6, 0x8f, 0x21, 0x56, 0x75, 0xa5, 0xdb, 0xba, 0xcb, 0x5f, 0x08, 0x00, 0x00, 0x00,
+];
+
+pub struct Instructions<T>
+where
+    T: Deref<Target = [u8]>,
+{
+    data: T,
+}
+
+impl<T> Instructions<T>
+where
+    T: Deref<Target = [u8]>,
+{
+    /// Creates a new `Instructions` struct.
+    ///
+    /// `data` is the instructions sysvar account data.
+    ///
+    /// # Safety
+    ///
+    /// This function is unsafe because it does not check if the provided data is from the Sysvar Account.
+    pub unsafe fn new(data: T) -> Self {
```

Comment:
Let's call this `new_unchecked` to maintain the same pattern for unsafe methods.